### DISCUSSION TITLE
Cognitive relay aided-networks dataset using Sionna RTX

### DISCUSSION THREAD

**Discussion by yacine074** [NVlabs/sionna#227] [General]
Hello,

I am interested in simulating a cognitive relay-aided network that includes a primary transmitter (UP) and its destination (DP), a secondary full-duplex relay (R), and a secondary user (US) and its destination (DS) as described in this [paper](https://hal.science/hal-03534545/document). To do this, I've created a [code](https://colab.research.google.com/drive/1KQSN0eqlZW_qdb4xLHZ94iC24Bp4JCnT?usp=sharing) that produces a dataset with the different channel gains between each user.

The problem I have is that the various channels are weak, with a mean equal to 0 and a standard deviation of 0.70, with a minimum equal to -1.78 and a maximum equal to 1.88.

What can I change to have channels between -30 and 30?

Is this message related to the scenario I'm using?

Thanks in advance, and best regards.

**Comment by jhoydis** [comment]
Hi,

For which quantity do you compute these values (mean, std, min, max)? Total received power? Coefficients of the channel impulse or frequency response?

If you want to have more control over the dynamic range, you can simply normalise all channel impulse responses and then scale them according to the desired path loss you would like to have.

**Comment by yacine074** [comment]
Hello,

Thank you for your response. The quantity for which I am computing these values is the frequency response of the channel.

So, is the idea as follows?

**1- Normalize the frequency response of the channel, which is a complex array:**

```

# h_PP.real: Assuming I am taking the frequency response of the channel of the primary transmitter/primary receiver
normalized_channel = h_PP.real/np.sum(h_PP.real)

```
**2- Then, scale the results according to the desired path loss:**


```
desired_path_loss = 40  # Replace this with your desired path loss value
scaled_channel = desired_path_loss * normalized_channel
```

**3-Then, I apply the same process to the imaginary part of the frequency response of the channel.**

**Reply by jhoydis** [reply]
Hi,

This is roughly correct, but not entirely.

First of all, when you generate the channel frequency response using the function [``cir_to_ofdm_channel``](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.cir_to_ofdm_channel), you need to turn on the flag ``normalize=True``. This ensures that the average energy across the resource grid is equal to one.

Then you would fix a path loss, typically in dB:

``pl = 60dB``

and apply it to the channel frequency response ``h_freq`` as follows:
```
h = tf.sqrt(10**(-pl/10))*h_freq
```
Note that the square-root is used.

However, by doing this manual scaling of the channels, you loose everything related to the relative positions of the devices.

**Reply by yacine074** [reply]
Thank your for your answer. Even after applying the modifications you suggested, the maximum value of h still remains low. For instance, if we set the path loss to 0dB, the maximum value of h is only 1.88.

**Reply by jhoydis** [reply]
Do you mean by maximum value, maximum channel gain across all subcarriers?
With path loss of 0dB, the average squared value of the channel gains is equal to one. So, 1.88 sounds plausible. 

Why do you care so much about the maximum value of h? What matters is the signal to noise ratio (SNR) that you see at the receiver.

**Reply by jhoydis** [reply]
Do you use ray tracing or stochastic channel models for this? Keep in mind, that unless you change the device positions, there is no randomness in the channels with ray tracing which might be not very interesting for link-level simulations.
